src/Fixer/PhpUnit/PhpUnitSetUpTearDownVisibilityFixer.php

```diff
@@ -0,0 +1,165 @@
+            $this->fixSetUpAndTearDown($tokens, $indexes[0], $indexes[1]);
+        }
+    }
+
+    /**
+     * @param Tokens $tokens
+     *
+     * @return int[][] array of [start, end] indexes from sooner to later classes
+     */
+    private function findPhpUnitClasses(Tokens $tokens)
+    {
+        $phpUnitTestCaseIndicator = new PhpUnitTestCaseIndicator();
+        $phpunitClasses = [];
+
+        for ($index = 0, $limit = $tokens->count() - 1; $index < $limit; ++$index) {
+            if ($tokens[$index]->isGivenKind(T_CLASS) && $phpUnitTestCaseIndicator->isPhpUnitClass($tokens, $index)) {
+                $index = $tokens->getNextTokenOfKind($index, ['{']);
+                $endIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_CURLY_BRACE, $index);
+                $phpunitClasses[] = [$index, $endIndex];
+                $index = $endIndex;
+            }
+        }
+
+        return $phpunitClasses;
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $startIndex
+     * @param int    $endIndex
+     */
+    private function fixSetUpAndTearDown(Tokens $tokens, $startIndex, $endIndex)
+    {
+        $tokensAnalyzer = new TokensAnalyzer($tokens);
+        for ($i = $endIndex - 1; $i > $startIndex; --$i) {
+            if (!$this->isMethod($tokens, $i)) {
+                continue;
+            }
+            if (!$this->isMethodSetUpOrTearDown($tokens, $i)) {
+                continue;
+            }
+
+            $visibility = $tokensAnalyzer->getMethodAttributes($i)['visibility'];
+
+            if (T_PROTECTED === $visibility) {
+                continue;
+            }
+
+            if (T_PUBLIC === $visibility) {
+                do {
+                    $i = $tokens->getPrevMeaningfulToken($i);
+                } while (!$tokens[$i]->isGivenKind(T_PUBLIC));
+
+                $tokens[$i] = new Token([T_PROTECTED, 'protected']);
+
+                continue;
+            }
+
+            if (null === $visibility) {
+                $tokens->insertAt($i, [new Token([T_PROTECTED, 'protected']), new Token([T_WHITESPACE, ' '])]);
+            }
+        }
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $index
+     *
+     * @return bool
+     */
+    private function isMethod(Tokens $tokens, $index)
+    {
+        $tokensAnalyzer = new TokensAnalyzer($tokens);
+
+        return $tokens[$index]->isGivenKind(T_FUNCTION) && !$tokensAnalyzer->isLambda($index);
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $index
+     *
+     * @return bool
+     */
+    private function isMethodSetUpOrTearDown(Tokens $tokens, $index)
+    {
+        $functionNameIndex = $tokens->getNextMeaningfulToken($index);
+        $functionName = \strtolower($tokens[$functionNameIndex]->getContent());
+
+        if ('setup' !== $functionName && 'teardown' !== $functionName) {
+            return false;
+        }
+
+        return true;
```

Comment:
can be one line `return 'setup' !== $functionName && 'teardown' !== $functionName;`